x-pack/plugins/cloud_integrations/cloud_chat/public/plugin.tsx

```diff
@@ -89,3 +95,12 @@ export class CloudChatPlugin implements Plugin {
     }
   }
 }
+
+const isTrialWindow = (buffer: number, trialEndDate?: Date) => {
```

Comment:
This could be extracted to `common`, so it can be reused on the server and the client.